File: algorithm/455. Assign Cookies/455.py
class Solution(object):
    def findContentChildren(self, g, s):
        """
        :type g: List[int]
        :type s: List[int]
        :rtype: int
        """
        if not g or not s:
            return 0
        g.sort()
        s.sort()
        i,j=0,0
        while i < len(g) and j < len(s):
            if g[i]<=s[j]:
                i+=1
            j+=1
        return i
        

File: algorithm/455. Assign Cookies/455.cpp
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(),g.end());
        sort(s.begin(),s.end());
        int result=0;
        for (int i=0; i <s.size()&&result<g.size();i++)
        {
            if(g[result]<=s[i]) result++;
        }
        return result;
    }
};
